crates/goose/src/providers/canonical/name_builder.rs

```diff
@@ -0,0 +1,869 @@
+    if model_lower.starts_with("grok") || model_lower.contains("grok") {
+        return Some("x-ai");
+    }
+
+    if model_lower.starts_with("jamba") || model_lower.contains("jamba") {
+        return Some("ai21");
+    }
+
+    if model_lower.starts_with("command") || model_lower.contains("command") {
+        return Some("cohere");
+    }
+
+    None
+}
+
+/// Strip common prefixes from model names using pattern matching
+/// Looks for known model family patterns and strips everything before them
+fn strip_common_prefixes(model: &str) -> String {
+    let model_patterns = [
+        "claude-",
+        "gpt-",
+        "gemini-",
+        "gemma-",
+        "o1-",
+        "o1",
+        "o3-",
+        "o3",
+        "o4-",
+        "meta-llama-",
+        "mistral-",
+        "mixtral-",
+        "chatgpt-",
+        "deepseek-",
+        "qwen-",
+        "grok-",
+        "jamba-",
+        "command-",
+        "codestral",
+        "ministral-",
+        "pixtral-",
+        "devstral-",
+    ];
+
+    let mut earliest_pos = None;
+
+    for pattern in &model_patterns {
+        if let Some(pos) = model.to_lowercase().find(pattern) {
+            if earliest_pos.is_none() || pos < earliest_pos.unwrap() {
+                earliest_pos = Some(pos);
+            }
+        }
+    }
+
+    // If we found a pattern, strip everything before it
+    if let Some(pos) = earliest_pos {
+        return model.get(pos..).unwrap_or(model).to_string();
+    }
+
+    model.to_string()
+}
+
+/// Try to extract provider prefix from model names like "databricks-meta-llama-3-1-70b"
+/// Returns (provider, model) tuple if found
+fn extract_provider_prefix(model: &str) -> Option<(&'static str, &str)> {
+    let known_providers = [
+        "anthropic",
+        "openai",
+        "google",
+        "meta-llama",
+        "mistralai",
+        "cohere",
+        "ai21",
+        "amazon",
+        "deepseek",
+        "qwen",
+        "x-ai",
+        "nvidia",
+        "microsoft",
+        "perplexity",
+    ];
+
+    for provider in &known_providers {
+        let prefix = format!("{}-", provider);
+        if model.starts_with(&prefix) {
+            if let Some(model_part) = model.strip_prefix(&prefix) {
+                return Some((provider, model_part));
+            }
+        }
+    }
+
+    None
+}
+
+/// Strip version suffixes from model names and normalize version numbers
+pub fn strip_version_suffix(model: &str) -> String {
+    let mut result = NORMALIZE_VERSION_RE
+        .replace_all(model, "-$1.$2$3")
+        .to_string();
+
+    let mut changed = true;
+    while changed {
+        let before = result.clone();
+        for pattern in STRIP_PATTERNS.iter() {
+            result = pattern.replace(&result, "").to_string();
+        }
+        changed = result != before;
+    }
+
+    result
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_canonical_names() {
+        // canonical_name tests
+        assert_eq!(
```

Comment:
again, very repetitive - use parameterized test. also this is an insane long list of tests. strip them down to the ones that we actually need.